test/test_nn.py

```diff
@@ -11526,6 +11527,136 @@ def __init__(self):
                 self.assertEqual(p.grad.to(devices[0]), pe.grad)
 
 
+class TestLazyModules(TestCase):
```

Comment:
You might want to make this test class device generic so it runs on both CPU and CUDA (and other device types, like XLA). If you're confident this will work properly on CUDA then that can also happen in the future.

Reply:
Is there a test for when an uninitialized param is registered on a module that doesn't know how to infer?

Reply:
Also: would a test that serializes and loads a state dict (from an IO buffer, for example) containing uninitialized values, be interesting?

Reply:
> You might want to make this test class device generic so it runs on both CPU and CUDA (and other device types, like XLA). If you're confident this will work properly on CUDA then that can also happen in the future.

Sounds good, let me look other tests to see how is this done 😅

> Is there a test for when an uninitialized param is registered on a module that doesn't know how to infer?

`test_lazy_module_parameter` sets uninitialized parameters in the generic nn.Module, which does not know how to infer. 

We expect that in this situation, the module will error in the forward pass or the `infer_parameters` as the parameters will remain uninitialized.

> Also: would a test that serializes and loads a state dict (from an IO buffer, for example) containing uninitialized values, be interesting?

Yes, I think I have to add this case

Reply:
I've been looking at making the class generic, but in all the modules the parameters are first created in the CPU and then moved to the corresponding device. I think that the current test that moves an uninitialized parameter to a gpu should be enough?

Reply:
Your call. If you're not worried about the inference path being device-dependent, for instance, I don't think you need to do it, especially not in your first PR.